ComfortZoneApp/app/src/main/java/com/example/comfortzone/fragments/CityListViewFragment.java

```diff
@@ -1,38 +1,26 @@
 package com.example.comfortzone.fragments;
 
 import android.os.Bundle;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
 
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 import androidx.fragment.app.Fragment;
 import androidx.recyclerview.widget.LinearLayoutManager;
 import androidx.recyclerview.widget.RecyclerView;
 
-import android.os.Parcel;
-import android.os.Parcelable;
-import android.util.Log;
-import android.view.LayoutInflater;
-import android.view.View;
-import android.view.ViewGroup;
-import android.widget.ArrayAdapter;
-
-import com.example.comfortzone.AllWeathersDatabase;
 import com.example.comfortzone.FlightsAdapter;
 import com.example.comfortzone.R;
+import com.example.comfortzone.UpdateCityListCallback;
 import com.example.comfortzone.models.WeatherData;
 
-import org.parceler.Parcels;
-
 import java.util.ArrayList;
 import java.util.List;
-import java.util.stream.Collectors;
 
-/**
- * A simple {@link Fragment} subclass.
- * Use the {@link CityListViewFragment#newInstance} factory method to
- * create an instance of this fragment.
- */
-public class CityListViewFragment extends Fragment {
+public class CityListViewFragment extends Fragment implements UpdateCityListCallback {
```

Comment:
ah that's smart! you are using interface instead, so you don't need to pass the list in through arguments, you will need to have a means to continuously update the list anyway, so creating new fragment each time wouldn't be ideal